department-of-veterans-affairs/caseflow
# Shalliburton/caseflow 1078 distribute hearing days evenly

Resolves [CASEFLOW-1078](https://vajira.max.gov/browse/CASEFLOW-1078)

### Description
Added a max allocations per day constraint to prevent assigning too many hearing days per date such that a given RO is unable to handle the workload.

### Acceptance Criteria
- [ ] When a user uploads the RO/CO hearings spreadsheet for a given date range, then the number of hearing days created for any date is close to the average (we can make that more specific)
- [ ] (total # of hearing days without rooms across all ROs) / (available days in the range) rounded up
- [ ] (total # of hearing days without rooms across all ROs) / (total # of weekdays in range - holidays - # of Board non-availability dates) rounded up

### Testing Plan
1. Shadow user `BVATWARNER` and navigate to the build schedule page: http://localhost:3000/hearings/schedule/build
2. Click the `Upload Files` button
3. Set the start date to 04/01/2021 and the end date to 06/30/2021
4. Locate the `invalidRoSpreadsheet.xlsx` file in `spec/support/invalidRoSpreadsheet.xlsx` and use that as the file to upload
5. Ensure that the algorithm ends in error and the rooms are not created
